redfish-core/lib/systems.hpp

```diff
@@ -1741,34 +1741,78 @@ inline void getPowerMode(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
                 messages::internalError(aResp->res);
                 return;
             }
-            // Valid Power Mode object found, now read the current value
+
+            // Valid Power Mode object found, now read the mode variables.
             crow::connections::systemBus->async_method_call(
                 [aResp](const boost::system::error_code ec,
-                        const std::variant<std::string>& pmode) {
+                        PropertiesType& properties) {
                     if (ec)
                     {
                         // Service not available, no error, return no data
                         BMCWEB_LOG_DEBUG
-                            << "Service not available on PowerMode Get: " << ec;
+                            << "Service not available on PowerMode GetAll: "
+                            << ec;
                         return;
                     }
 
-                    const std::string* s = std::get_if<std::string>(&pmode);
-                    if (s == nullptr)
+                    BMCWEB_LOG_DEBUG << "Size: " << properties.size();
+
+                    for (const auto& property : properties)
                     {
-                        BMCWEB_LOG_DEBUG << "Unable to get PowerMode value";
-                        messages::internalError(aResp->res);
-                        return;
-                    }
+                        if (property.first == "PowerMode")
+                        {
+                            const std::string* s =
+                                std::get_if<std::string>(&property.second);
+
+                            if (s == nullptr)
+                            {
+                                BMCWEB_LOG_DEBUG
+                                    << "Unable to get PowerMode value ";
+                                messages::internalError(aResp->res);
+                            }
+                            else
```

Comment:
More typical of bmcweb code is return on the if; don't have an else 
E.g. 

                            if (s == nullptr)
                            {
                                BMCWEB_LOG_DEBUG
                                    << "Unable to get PowerMode value ";
                                messages::internalError(aResp->res);
                                return
                            }

Reply:
There are two pieces of data here.. powerMode, and safeMode.
If you return out of one of them, you will never try to get the other one.
-
How often would this happen, getting the first property, but not getting the second property ?
I can put the return in.   but seems odd to me.